context/app/static/js/components/Header/Menu/Menu.jsx

```diff
@@ -32,28 +32,19 @@ function Menu(props) {
               <DropdownLink key={type} href={`/search?entity_type[0]=${type}`}>{`${type}s`}</DropdownLink>
             ))}
             <DropdownLink href="/collections">Collections</DropdownLink>
-
-            {/*
-                If this changes, remember to update HeaderContent.jsx!
-            */}
-
-            <DropdownMenuItem onClick={togglePreview}>
-              Previews
-              {openPreview ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
-            </DropdownMenuItem>
-            {openPreview && <PreviewLinks isIndented />}
-
-            <DropdownMenuItem onClick={toggleCCF}>
-              Atlas & Tools
-              {openCCF ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
-            </DropdownMenuItem>
-            {openCCF && <CCFLinks isIndented />}
-
-            <DropdownMenuItem onClick={toggleDocumentation}>
-              Documentation
-              {openDocumentation ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
-            </DropdownMenuItem>
-            {openDocumentation && <DocumentationLinks isIndented />}
+            {[
+              [toggleResources, 'Resources', openResources, ResourceLinks],
+              [toggleAtlasTools, 'Atlas & Tools', openAtlasTools, AtlasToolsLinks],
+              [toggleDocumentation, 'Documentation', openDocumentation, DocumentationLinks],
+            ].map(([onClick, label, isOpen, Component]) => (
+              <>
+                <DropdownMenuItem onClick={onClick}>
+                  {label}
+                  {isOpen ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
+                </DropdownMenuItem>
+                {isOpen && <Component isIndented />}
+              </>
+            ))}
```

Comment:
We could probably create a new component here like `Submenu` and move the open state into the component and only pass the string and links components as props.